contentprovider.go

```diff
@@ -715,3 +716,31 @@ func sortChunkMatchesByScore(ms []ChunkMatch) {
 func SortFilesByScore(ms []FileMatch) {
 	sort.Sort(fileMatchSlice(ms))
 }
+
+type rankedFileMatchSlice []FileMatch
+
+const epsilon = 0.00000001
+
+func (m rankedFileMatchSlice) Len() int      { return len(m) }
+func (m rankedFileMatchSlice) Swap(i, j int) { m[i], m[j] = m[j], m[i] }
+func (m rankedFileMatchSlice) Less(i, j int) bool {
+	r1 := m[i].ranks
+	r2 := m[j].ranks
```

Comment:
can we swap out ranks for a constructed score if they are empty, so we can combine stuff in prod?

Reply:
something like one, but maybe downscaled.